.github/workflows/test.yml

```diff
@@ -248,6 +248,10 @@ jobs:
         with:
           submodules: true
           token: ${{ secrets.SUBMODULES_TOKEN }}
+        name: Checkout main on all submodules
+        run: |
+          git submodule foreach git checkout main
+          git submodule foreach git pull origin main
```

Comment:
Should we create a dedicated action to checkout with the submodules for all the workflows? Or will just be needed for the E2E enterprise tests? I feel like we might need to check it out for all of them but might miss any nuance that is not needed all the time.

Reply:
for not i am only testing here, and yes we will need it in couple of places